tests/grid/grid_generator_half_hyper_shell.cc

```diff
@@ -0,0 +1,83 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Test boundary ids for GridGenerator::half_hyper_shell()
+
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include "../tests.h"
+
+
+
+void
+dim_2()
+{
+  Triangulation<2> tr;
+  double           r_inner = 1.;
+  double           r_outer = 2.;
+
+  GridGenerator::half_hyper_shell(
+    tr, Point<2>(0.0, 0.0), r_inner, r_outer, 0, true);
+
+  Triangulation<2>::cell_iterator cell = tr.begin();
+  for (; cell != tr.end(); ++cell)
+    {
+      for (unsigned int f = 0; f < 4; ++f)
+        {
+          if (cell->face(f)->at_boundary())
+            deallog << "half_hyper_shell<2>:: cell " << cell->index()
+                    << ", face " << f << ", id " << cell->face(f)->boundary_id()
+                    << std::endl;
+        }
+    }
+}
+
+void
+dim_3()
+{
+  Triangulation<3> tr;
+  double           r_inner = 1.;
+  double           r_outer = 2.;
+
+  GridGenerator::half_hyper_shell(
+    tr, Point<3>(0.0, 0.0, 0.0), r_inner, r_outer, 0, true);
+
+  Triangulation<3>::cell_iterator cell = tr.begin();
+  for (; cell != tr.end(); ++cell)
+    {
+      for (unsigned int f = 0; f < 6; ++f)
+        {
+          if (cell->face(f)->at_boundary())
+            deallog << "half_hyper_shell<3>:: cell " << cell->index()
+                    << ", face " << f << ", id " << cell->face(f)->boundary_id()
+                    << std::endl;
+        }
+    }
+}
+
+
+int
+main()
+{
+  initlog();
+  deallog.depth_console(0);
```

Comment:
this line is not needed.

Reply:
which one? the `deallog`?

Reply:
yes, sorry for not being more precise.

Reply:
No worries! When I read code reviews I sometimes forget whether the top or bottom line is the one being reviewed. I guess it wouldn't make sense to say `main()` wasn't needed here so that's on me :stuck_out_tongue: